city_metrix/layers/layer_geometry.py

```diff
@@ -0,0 +1,291 @@
+    miny = float(math.floor(min_y))
+    maxx = float(math.floor(max_x))
+    maxy = float(math.floor(max_y))
+
+    bounds = (minx, miny, maxx, maxy)
+
+    ee_rectangle = ee.Geometry.Rectangle(
+        [minx, miny, maxx, maxy],
+        crs,
+        geodesic=False
+    )
+
+    rectangle = {"ee_geometry": ee_rectangle, "crs": crs, "bounds": bounds}
+    return rectangle
+
+
+def reproject_units(min_x, min_y, max_x, max_y, from_crs, to_crs):
+    """
+    Project coordinates from one map EPSG projection to another
+    """
+    from pyproj import Transformer
+    transformer = Transformer.from_crs(from_crs, to_crs)
+
+    # Note: order of coordinates must be lat/lon
+    sw_coord = transformer.transform(min_x, min_y)
+    ne_coord = transformer.transform(max_x, max_y)
+
+    reproj_min_x = float(sw_coord[0])
+    reproj_min_y = float(sw_coord[1])
+    reproj_max_x = float(ne_coord[0])
+    reproj_max_y = float(ne_coord[1])
+
+    reproj_bbox = (reproj_min_y, reproj_min_x, reproj_max_y, reproj_max_x)
+    return reproj_bbox
+
+
+def get_utm_zone_from_latlon_point(centroid) -> str:
+    """
+    Get the UTM zone projection for given a bounding box.
+
+    :param bbox: tuple of (min x, min y, max x, max y)
+    :return: the EPSG code for the UTM zone of the centroid of the bbox
+    """
+
+    utm_x, utm_y, band, zone = utm.from_latlon(centroid.y, centroid.x)
+
+    if centroid.y > 0:  # Northern zone
+        epsg = 32600 + band
+    else:
+        epsg = 32700 + band
+
+    return f"EPSG:{epsg}"
+
+
+def create_fishnet_grid(bbox:GeoExtent,
+                        tile_side_length:float=0, tile_buffer_size:float=0, length_units:str="meters",
+                        spatial_resolution:int=1, output_as:str="utm"):
+    # NOTE: the AOI can be specified in either WGS or UTM, but the generated tile grid is always in UTM
+
+    tile_side_length = 0 if tile_side_length is None else tile_side_length
+    tile_buffer_size = 0 if tile_buffer_size is None else tile_buffer_size
+    spatial_resolution = 1 if spatial_resolution is None else spatial_resolution
+
+    if length_units is None:
+        if (tile_side_length > 0 or tile_buffer_size > 0):
+            raise Exception("Length_units cannot be None if tile_side_length or tile_buffer_size are > 0")
+        else:
+            length_units = "meters" # a placeholder value
+
+    length_units = length_units.lower()
+    output_as = 'utm' if output_as is None else output_as.lower()
+
+    valid_length_units = ['degrees', 'meters']
+    if length_units not in valid_length_units:
+        raise ValueError(f"Invalid length_units ('{length_units}'). "
+                         f"Valid methods: [{valid_length_units}]")
+
+    valid_output = ['utm', 'geographic']
+    if output_as not in valid_output:
+        raise ValueError(f"Invalid grid_units ('{output_as}'). "
+                         f"Valid methods: [{valid_output}]")
+
+    if length_units == "degrees":
+        if tile_side_length < 0.001:
```

Comment:
Could you pull these limits out to constants with maybe some comment description on how they were determined, in case someone needs to change them in the future?